packages/plugin/automatic-persisted-queries/src/plugin.ts

```diff
@@ -168,19 +168,19 @@ export const ezAutomaticPersistedQueries = (options?: AutomaticPersistedQueryOpt
 
         options.query = query;
 
-        // override execute so we can store query if it's valid
-        options.execute = async (...args: any[]) => {
-          const result: ExecutionResult = await execute(...args);
+        // store query if it's valid
+        options.validate = (...args) => {
+          const errors: readonly GraphQLError[] | null = validate(...args);
 
-          if (!result.errors?.length) {
+          if (!errors?.length) {
             try {
```

Comment:
AFAIR, Promise.resolve has an implicit `try/catch`, so the additional block should not be needed.

Reply:
Nevermind. It's a direct call so it call indeed throw in the global context.

Reply:
yep 👍